Simplified/Book/UI/NYPLBookCellDelegate.m

```diff
@@ -413,33 +425,50 @@ - (void)audiobookManagerDidRequestRefresh {
   [[NYPLBookRegistry sharedRegistry] setState:NYPLBookStateDownloadNeeded forIdentifier:self.book.identifier];
 
 #if FEATURE_OVERDRIVE
-  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(updateODAudiobookManifest) name:NSNotification.NYPLMyBooksDownloadCenterDidChange object:nil];
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(updateODAudiobookManifest)
+                                               name:NSNotification.NYPLMyBooksDownloadCenterDidChange
+                                             object:nil];
 #endif
   [[NYPLMyBooksDownloadCenter sharedDownloadCenter] startDownloadForBook:self.book];
 }
 
 #if FEATURE_OVERDRIVE
 - (void)updateODAudiobookManifest {
-  if ([[NYPLBookRegistry sharedRegistry] stateForIdentifier:self.book.identifier] == NYPLBookStateDownloadSuccessful) {
+  NYPLBookState bookState = [[NYPLBookRegistry sharedRegistry]
+                             stateForIdentifier:self.book.identifier];
+
+  if (bookState == NYPLBookStateDownloadFailed) {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+    [self.refreshAudiobookLock unlock];
+    return;
+  }
+
+  if (bookState == NYPLBookStateDownloadSuccessful) {
+    if (![(NSObject*)self.manager.audiobook isKindOfClass:[OverdriveAudiobook class]]) {
```

Comment:
this is kind of a patch because from the stack trace it looks like we get in here even when `self.manager.audiobook` is a `NYPLAEToolkit.FindawayAudiobook` object, so not just for `OverdriveAudiobook`. I'm not sure why this is happening (I actually don't know yet how to diagnose this) but this guard seemed necessary. However, should we remove the observer? and should we unlock?

Reply:
Yeah, I think removing the observer and unlock is the right thing to do here.

Do you think the check `[book.distributor isEqualToString:OverdriveDistributorKey]` would help to avoid audiobooks from other distributers getting into this part of code. If so, we can add one at the beginning of this method and one where we added the observer.